src/sinks/util/service.rs

```diff
@@ -84,76 +85,110 @@ impl<L> ServiceBuilderExt<L> for ServiceBuilder<L> {
 /// Middleware settings for outbound requests.
 ///
 /// Various settings can be configured, such as concurrency and rate limits, timeouts, etc.
+#[serde_as]
 #[configurable_component]
 #[derive(Clone, Copy, Debug)]
 pub struct TowerRequestConfig {
     #[configurable(derived)]
-    #[serde(default)]
+    #[serde(default = "default_concurrency")]
     #[serde(skip_serializing_if = "concurrency_is_none")]
     pub concurrency: Concurrency,
 
-    /// The maximum time a request can take before being aborted.
+    /// The time a request can take before being aborted.
     ///
     /// It is highly recommended that you do not lower this value below the service’s internal timeout, as this could
     /// create orphaned requests, pile on retries, and result in duplicate data downstream.
+    #[configurable(metadata(docs::type_unit = "seconds"))]
+    #[serde(default = "default_timeout_secs")]
     pub timeout_secs: Option<u64>,
 
-    /// The time window, in seconds, used for the `rate_limit_num` option.
+    /// The time window used for the `rate_limit_num` option.
+    #[configurable(metadata(docs::type_unit = "seconds"))]
+    #[serde(default = "default_rate_limit_duration_secs")]
     pub rate_limit_duration_secs: Option<u64>,
 
     /// The maximum number of requests allowed within the `rate_limit_duration_secs` time window.
+    #[configurable(metadata(docs::type_unit = "requests"))]
+    #[serde(default = "default_rate_limit_num")]
     pub rate_limit_num: Option<u64>,
 
     /// The maximum number of retries to make for failed requests.
     ///
     /// The default, for all intents and purposes, represents an infinite number of retries.
+    #[configurable(metadata(docs::type_unit = "retries"))]
+    #[serde(default = "default_retry_attempts")]
     pub retry_attempts: Option<usize>,
 
-    /// The maximum amount of time, in seconds, to wait between retries.
+    /// The maximum amount of time to wait between retries.
+    #[configurable(metadata(docs::type_unit = "seconds"))]
+    #[serde(default = "default_retry_max_duration_secs")]
     pub retry_max_duration_secs: Option<u64>,
 
-    /// The amount of time to wait before attempting the first retry for a failed request.
+    /// The maximum amount of time to wait before attempting the first retry for a failed request.
```

Comment:
I don't think this is a maximum?
```suggestion
    /// The amount of time to wait before attempting the first retry for a failed request.
```